koku/masu/processor/expired_data_remover.py

```diff
@@ -151,4 +154,7 @@ def remove(self, simulate=False, provider_uuid=None, line_items_only=False):
                 removed_data = self._cleaner.purge_expired_line_item(expired_date=expiration_date, simulate=simulate)
             else:
                 removed_data = self._cleaner.purge_expired_report_data(expired_date=expiration_date, simulate=simulate)
+            # Separately remove expired CostUsageReportManifests
+            CostUsageReportManifest.objects.filter(billing_period_start_datetime__lt=expiration_date).delete()
```

Comment:
So, what you have here probably works right now because I have not finished part 2 of my issue to add a trigger for the line items only data removal here: https://github.com/project-koku/koku/issues/1616

However, when I do this `.remove()` method will be called for both when we delete a report, but also for when we want to clean out the line items raw data in the database. Which means now with this way this is organized whenever we delete the raw line item data we will also be deleting the `CostUsageReportManifest` which for sure doesn't sound like what we want. I think we may want to move this at the very least under: https://github.com/project-koku/koku/pull/1788/files#diff-ec0942de17133f6841f909b02e083437R156

**Side Question**: 

So, the way that this method is dynamically created the `self._cleaner.purge_expired_report_data` will actually be called based off the provider type. So, this line will actually trigger one of the following:
- https://github.com/project-koku/koku/blob/master/koku/masu/processor/aws/aws_report_db_cleaner.py#L98
- https://github.com/project-koku/koku/blob/master/koku/masu/processor/ocp/ocp_report_db_cleaner.py#L94
- https://github.com/project-koku/koku/blob/master/koku/masu/processor/azure/azure_report_db_cleaner.py#L97

So, I am not as familiar with the `CostUsageReportManifest`, but I think this is deleting all manifest before the start date regardless of provider type, every time a report is deleted. Is that what we want?  If not we may want to set the provider, flag here too:
```
CostUsageReportManifest.objects.filter(provider=model, billing_period_start_datetime__lt=expiration_date)
```
If we are gonna group by provider, it may be better to move this into each `purge_expired_report_data` method I linked above since it is already being called base on provider report deleting. That would allow you to lean on some of the logic already in those methods too.

Reply:
whooops, sorry @nbon12, I started reviewing before I saw you move it back to WIP. Hopefully my comments are helpful though.

Reply:
@myersCody how about we separate the deletion of the expired CostUsageReportManifests into a separate task?

Reply:
@myersCody regarding the side question, the answer is yes; we want to delete all CostUsageReportrManifests regardless of the provider type, according to @adberglund

Reply:
@nbon12 @myersCody So looking at the way we call this, we only ever call the expired data remover for 1 provider at a time so when we do remove the manifest objects we'll want to do what Cody suggested and filter on the provider.

Reply:
Hmm. I think at the very least we want to be careful to not delete the manifests when line items only raw data is being deleted. The thought being just cause we removed the raw data cause it was summarized, doesn't mean that we have deleted the report itself from the system. We should add this as a separate method inside of the `ExpiredDataRemover`, and just make sure to only call it when line_items_only is false inside of the .remove() method. Additionally, I think that since ExpiredDataRemover gets called for a single provider at a time, we should just go ahead and only delete a single provider at a time. I reached out to Andrew and he agrees that it makes sense to make the removal of the manifest provider based since it will just live in this class anyway. Also, we can do a quick bluejeans to talk about this too if you would like. I feel like I have some background in the line items only stuff since I picked up that issue that would probably be helpful to know here.

Reply:
@myersCody added